hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/RDDCustomColumnsSortPartitioner.java

```diff
@@ -43,14 +43,14 @@
 
   public RDDCustomColumnsSortPartitioner(HoodieWriteConfig config) {
     this.serializableSchema = new SerializableSchema(new Schema.Parser().parse(config.getSchema()));
-    this.sortColumnNames = getSortColumnName(config);
+    this.sortColumnNames = BulkInsertPartitioner.prependPartitionPathColumn(getSortColumnName(config), config);
```

Comment:
We need to make sure we only prepend for partitioned tables

Reply:
I thought that checking for the empty config in BulkInsertPartitioner accomplished that

Reply:
It will be pre-empted by meta-fields conditional (if these are enabled)

We can add that check into `prependPartitionPathColumn` itself, but in that case let's rename it accordingly to  make such contingency clear: `tryPrependPartitionColumns`